crates/revm/src/executor.rs

```diff
@@ -333,24 +345,31 @@ where
                     cumulative_gas_used += result.gas_used()
                 }
 
-                // Route the l1 cost and base fee to the appropriate optimism vaults
-                self.increment_account_balance(
-                    block.number,
-                    optimism::l1_cost_recipient(),
-                    l1_cost,
-                    &mut post_state,
-                )?;
-                self.increment_account_balance(
-                    block.number,
-                    optimism::base_fee_recipient(),
-                    U256::from(
-                        block
-                            .base_fee_per_gas
-                            .unwrap_or_default()
-                            .saturating_mul(result.gas_used()),
-                    ),
-                    &mut post_state,
-                )?;
+                if self.chain_spec.optimism.is_some() {
+                    match l1_cost {
+                        Some(l1_cost) => {
+                            // Route the l1 cost and base fee to the appropriate optimism vaults
```

Comment:
please move this comment above the `if self.chain_spec.optimism` line, as it explains both increments